includes/connection/class-wc-terms.php

```diff
@@ -51,7 +51,7 @@ public static function register_connections() {
 										'resolve'       => function( $source, array $args, AppContext $context, ResolveInfo $info ) use ( $tax_object ) {
 											$resolver = new TermObjectConnectionResolver( $source, $args, $context, $info, $tax_object->name );
 											$taxonomy = str_replace( ' ', '_', strtolower( $tax_object->labels->singular_name ) );
-											$resolver->set_query_arg( 'term_taxonomy_id', $source->{"{$taxonomy}_ids"} );
+											$resolver->set_query_arg( 'term_taxonomy_id', $source->attributes[$tax_object->name]['options'] );
```

Comment:
```suggestion
                                            $options = $source->attributes[ $tax_object->name ]['options'];
											$resolver->set_query_arg( 'term_taxonomy_id', ! empty( $options ) ? $options : array( '0' ) );
```
@jacobarriola This would ensure that no terms are returned for products that don't have that attribute assigned.